types/arcgis-js-api/tslint.json

```diff
@@ -9,6 +9,7 @@
         "no-any-union": false,
         "no-empty-interface": false,
         "no-irregular-whitespace": false,
+        "no-unnecessary-class": false,
```

Comment:
You would need to fix the errors from this or disable at the given line instead of disabling it for the whole package.